backend/src/main/java/com/zeros/backend/models/RestaurantAffiliate.java

```diff
@@ -0,0 +1,37 @@
+package com.zeros.backend.models;
+
+import java.time.LocalDateTime;
+
+public class RestaurantAffiliate extends Entity {
+
+    private final Long restaurantId;
+    private LocalDateTime affiliateDate;
+    private final Long qrCode;
+
+
+    public RestaurantAffiliate(Long restaurantId, Long qrCode, LocalDateTime affiliateDate) {
+        this.restaurantId = restaurantId;
+        this.qrCode = qrCode;
+        this.affiliateDate = affiliateDate;
+    }
+
+    public Long getRestaurantId() {
+        return restaurantId;
+    }
+
+    public LocalDateTime getAffiliateDate() {
+        return affiliateDate;
+    }
+
+    public Long getQrCode() {
+        return qrCode;
+    }
+
+    public boolean isSigned() {
+        return affiliateDate.isEqual(LocalDateTime.now()) || affiliateDate.isBefore(LocalDateTime.now());
+    }
+
+    public void signe() {
```

Comment:
@JumperSpirit what does signe stands for ?

Reply:
In my opinion the restaurant should subscribe and signe the affiliation by sending qrCode after receiving confirmation mail